x/concentrated-liquidity/swaps.go

```diff
@@ -20,6 +20,24 @@ type SwapState struct {
 	sqrtPrice                sdk.Dec // new current price when swap is done
 	tick                     sdk.Int // new tick when swap is done
 	liquidity                sdk.Dec // new liquidity when swap is done
+	feeGrowthGlobal          sdk.Dec // global fee growth per-swap
+}
+
+// updateFeeGrowthGlobal updates the swap state's fee growth global per unit of liquidity
+// when liquidity is positive.
+//
+// If the liquidity is zero, this is a no-op. This case may occur when there is no liquidity
+// between the ticks.This is possible when there are only 2 positions with no overlapping ranges.
+// As a result, the range from the end of position one to the beginning of position
+// two has no liquidity and can be skipped.
+// TODO: test
+func (ss *SwapState) updateFeeGrowthGlobal(feeChargeTotal sdk.Dec) {
+	if !ss.liquidity.IsZero() {
+		feeChargePerUnitOfLiquidity := feeChargeTotal.Quo(ss.liquidity)
+		// TODO: remove print.
+		fmt.Println("feeChargePerUnitOfLiquidity", feeChargePerUnitOfLiquidity)
+		ss.feeGrowthGlobal = ss.feeGrowthGlobal.Add(feeChargePerUnitOfLiquidity)
+	}
```

Comment:
This might change as we implement the swap in the other direction. I don't think it makes sense to test these right now given the high likelihood of these methods being changed / removed in the next PR.

Reply:
Also, these are tracked here:
- https://github.com/osmosis-labs/osmosis/issues/4089
- https://github.com/osmosis-labs/osmosis/issues/4086